### ISSUE TITLE
Selection of verification indicators

### ISSUE THREAD

**Issue by cainiao123s** [ultralytics/ultralytics#20008]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

In YOLO object detection, the final verification accuracy is inconsistent with the best verification accuracy during training. Which one should be taken as the standard?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cainiao123s, thank you for reaching out and for your interest in Ultralytics 🚀! 

We recommend checking out the [Docs](https://docs.ultralytics.com/) for answers to common questions, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, as well as guidance on evaluation metrics and training best practices.

If this is a ❓ Question about custom training or evaluation, please provide as much detail as possible, including the specific YOLO version you're using, dataset examples, and any relevant training logs. This will help us better understand your situation. You can also review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal performance.

If this is related to a 🐛 Bug, we kindly ask you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can investigate further.

Join the vibrant Ultralytics community where it suits you best! For real-time discussions, connect on [Discord](https://discord.com/invite/ultralytics) 🎧. You can also join detailed threads on [Discourse](https://community.ultralytics.com/) or explore our [Subreddit](https://reddit.com/r/Ultralytics) for knowledge sharing.

## Upgrade

Please ensure you're using the latest `ultralytics` package to avoid issues that may already be resolved. Upgrade all dependencies in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) using the following command:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following verified environments (with all required dependencies preinstalled, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/)):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI verifies the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to guide you. Rest assured, an Ultralytics engineer will review your issue and provide additional assistance soon 😊.

**Comment by Y-T-G** [comment]
How is it inconsistent?

**Comment by cainiao123s** [comment]
> How is it inconsistent?

![Image](https://github.com/user-attachments/assets/a21c142a-a3e9-4dd8-9758-0fafa75dfc1b)

**Comment by Y-T-G** [comment]
Best isn't based on mAP50 alone.

https://github.com/ultralytics/ultralytics/issues/16358#issuecomment-2360276418

**Comment by cainiao123s** [comment]
May I ask, in the process of writing a SCI paper, do you think it is better to use fitness as the criterion or map50 as the criterion, or should one choose an appropriate indicator by oneself?

> Best isn't based on mAP50 alone. It's based on fitness.
> 
> [#16358 (comment)](https://github.com/ultralytics/ultralytics/issues/16358#issuecomment-2360276418)

**Comment by cainiao123s** [comment]
> For academic papers, we recommend selecting evaluation metrics that align with your specific research objectives and community standards. While Ultralytics models use a [weighted fitness score](https://docs.ultralytics.com/reference/utils/metrics/#metrics.Metric.fitness) combining multiple metrics for internal optimization, scientific publications typically emphasize specific metrics like mAP50 or mAP50-95 depending on the task rigor required. Consider:
> 
> 1. **mAP50** if prioritizing detection recall with moderate localization precision
> 2. **mAP50-95** for strict localization accuracy requirements
> 3. **Domain conventions** - follow metrics used in comparable publications in your field
> 
> The choice should be explicitly justified in your methodology section based on your paper's focus. For comprehensive analysis, many papers report both metrics alongside class-specific AP values.

Thank you for your reply. Actually, what I want to understand in depth is: in research papers, MAP50 and MAP50-95 are generally used as indicators. But should I choose the MAP50 and MAP50-95 of the round with the best fitness, or the MAP50 and MAP50-95 of the round with the best MAP50?

**Comment by glenn-jocher** [comment]
@cainiao123s for research papers, we recommend selecting metrics from the epoch with the **best mAP50-95** (which typically aligns with best fitness since fitness weights mAP50-95 at 90% as shown in the [fitness calculation](https://docs.ultralytics.com/reference/utils/metrics/#metrics.Metric.fitness)). However, if your study specifically focuses on moderate-IoU detection performance, you might prioritize the best mAP50 epoch. Always explicitly state your selection criteria in the methodology section to ensure reproducibility.

**Comment by cainiao123s** [comment]
> [@cainiao123s](https://github.com/cainiao123s) for research papers, we recommend selecting metrics from the epoch with the **best mAP50-95** (which typically aligns with best fitness since fitness weights mAP50-95 at 90% as shown in the [fitness calculation](https://docs.ultralytics.com/reference/utils/metrics/#metrics.Metric.fitness)). However, if your study specifically focuses on moderate-IoU detection performance, you might prioritize the best mAP50 epoch. Always explicitly state your selection criteria in the methodology section to ensure reproducibility.
Thank you for your reply.

**Comment by glenn-jocher** [comment]
Acknowledged, glad we could help clarify! Let us know if you have any more questions.

**Comment by cainiao123s** [comment]
In subsequent research, I found that the final results obtained from best.pt, whether it was map50 or map50-95, were inconsistent with all previous epochs. The model parameters during validation were slightly lower than those during training, so I suspect that some minor adjustments were made to the model during validation.

<img width="1243" height="415" alt="Image" src="https://github.com/user-attachments/assets/d3f2e875-5a11-47a5-b0f9-dfe7f61edabb" />

<img width="1436" height="1068" alt="Image" src="https://github.com/user-attachments/assets/6cd549a5-ffeb-450e-b78e-df2592700f5f" />

**Comment by glenn-jocher** [comment]
Useful follow-up: `best.pt` is just the saved weights from the best training epoch—no extra “validation-only” changes are applied—so if a separate `yolo val` run gives lower mAP than the training log it’s almost always due to different validation settings or version; please re-run validation with the exact args from `runs/detect/train/args.yaml` (see the [Val mode docs](https://docs.ultralytics.com/modes/val/) for the relevant options), for example:

```bash
yolo val model=runs/detect/train/weights/best.pt data=... imgsz=... batch=... conf=... iou=...
```

and if the mismatch still exists, open a new issue with that command, your `results.csv`, and `yolo checks` output so we can investigate with better visibility.
